add warning flags

It would be great if we could add these flags to ig-sw: `'-Wall', '-Wextra', '-pedantic'` (as seen [here](https://github.com/psathyrella/partis/blob/2e962716f0649f3529047a4093dffd99191fef1a/packages/ig-sw/src/ig_align/SConstruct).

You sounded like you're familiar enough with the code to try -- the signed/unsigned comparison warnings are potentially scary, but fixing them also terrifies me since I don't know anything about the code...

```
gcc -o ig_align.o -c -std=gnu99 -Ofast -Wall -Wextra -pedantic -I. -Itclap ig_align.c
ig_align.c:61:26: warning: ISO C does not allow extra ';' outside of a function [-Wpedantic]
 KSEQ_INIT(gzFile, gzread);
                          ^
In file included from ig_align.c:15:0:
ig_align.c: In function 'ks_sample_cdec_score':
ksort.h:256:10: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
    if (k != n - pop - 1) tmp = a[k], a[k] = a[n-pop-1], a[n-pop-1] = tmp; \
          ^
ig_align.c:127:1: note: in expansion of macro 'KSORT_INIT'
 KSORT_INIT(cdec_score, aln_t, __aln_score_lt)
 ^
ig_align.c: In function 'align_read':
ig_align.c:247:23: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     for (int i = 0; i < kv_size(result); i++)
                       ^
ig_align.c: In function 'write_sam_records':
ig_align.c:328:22: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (a.loc.qe + 1 != read->seq.l)
                      ^
gcc -o ksw.o -c -std=gnu99 -Ofast -Wall -Wextra -pedantic -I. -Itclap ksw.c
ksw.c: In function 'push_cigar':
ksw.c:464:26: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  if (*n_cigar == 0 || op != (cigar[(*n_cigar) - 1]&0xf)) {
                          ^
gcc -o kstring.o -c -std=gnu99 -Ofast -Wall -Wextra -pedantic -I. -Itclap kstring.c
kstring.c: In function 'kvsprintf':
kstring.c:15:12: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
  if (l + 1 > s->m - s->l) {

```

--- a/src/ig_align/SConstruct
+++ b/src/ig_align/SConstruct
@@ -1,6 +1,6 @@
 env = Environment(
     CPPDEFINES=[],
-    CFLAGS = ['-std=gnu99', '-Ofast'],
+    CFLAGS = ['-std=gnu99', '-Ofast', '-Wall', '-Wextra', '-pedantic'],
     CPPPATH=[".", "tclap"])
 env.Program('ig-sw',
             [ "ig_align.c","ksw.c", "kstring.c", 'ig_align_main.cpp'],

--- a/src/ig_align/ig_align.c
+++ b/src/ig_align/ig_align.c
@@ -58,7 +58,7 @@
    (x) |= (x) >> 16, ++(x))
 #endif
 
-KSEQ_INIT(gzFile, gzread);
+KSEQ_INIT(gzFile, gzread)
 
 typedef kvec_t(kseq_t) kseq_v;
 
@@ -244,7 +244,8 @@ static aln_v align_read(const kseq_t *read, const kseq_v targets,
    * abort.
    */
   if (max_score < conf->min_score) {
-    for (int i = 0; i < kv_size(result); i++)
+    // kv_size returns the n field of a kvec_t, which is a size_t.
+    for (size_t i = 0; i < kv_size(result); i++)
       free(kv_A(result, i).cigar);
     kv_size(result) = 0;
     free(qry);
@@ -325,7 +326,12 @@ static void write_sam_records(kstring_t *str, const kseq_t *read,
         ksprintf(str, "D");
     }
 
-    if (a.loc.qe + 1 != read->seq.l)
+    // Here we explicitly cast the size_t l to be an int.
+    // Neither of these numbers will be near their maximum
+    // in our application, which even for a signed 4 byte int
+    // is 2147483647, and size_t is an unsigned 4 byte int in
+    // on a 32 bit machine.
+    if (a.loc.qe + 1 != (int)read->seq.l)
       ksprintf(str, "%luS", read->seq.l - a.loc.qe - 1);
 
     ksprintf(str, "\t*\t0\t0\t");

--- a/src/ig_align/ksort.h
+++ b/src/ig_align/ksort.h
@@ -253,7 +253,9 @@ typedef struct {
 			double z = 1., x = drand48(); \
 			type_t tmp; \
 			while (x < z) z -= z * i / (pop--); \
-			if (k != n - pop - 1) tmp = a[k], a[k] = a[n-pop-1], a[n-pop-1] = tmp; \
+			// As described in ig_align these numbers are not close to their max in our
+			// application, so we don't lose anything casting a size_t to an int.
+			if (k != (int)n - pop - 1) tmp = a[k], a[k] = a[n-pop-1], a[n-pop-1] = tmp; \
 			++k; \
 		} \
 	}

